### PR TITLE
feat(filters): allow filtering of tasks and reports in statistics

### PR DESCRIPTION
**Pull request by winged**
When filtering statistics, we face the problem that we want to
have a list of all tasks, and count the reported hours. Due to the
way Django works, the Sum of reported hours would trigger an (unfiltered)
subquery or Join, so our filters don't apply.

Therefore, we now have a "split" pseudo-queryset, which contains separate
querysets for tasks and reports. It then applies the filters to EITHER
of the querysets as needed (depending on prefix), then combines them
at the end of the filtering phase, using a subquery.

### CODE REVIEW

**Review by winged — COMMENTED** [review]
Looking good, but you need to find another reviewer, as I cannot approve it, coming from my fork :-)

**Review by fugal-dy — APPROVED** [review]
Nice work!

**Comment by trowik** [comment]
messed up commit history, so I opened a new PR #891

**Pull request closed without merging**
